veps/sig-network/decouple-nad.md

```diff
@@ -0,0 +1,229 @@
+
+Examples:
+
+- In case of SR-IOV, sriov-cni fails as it does not receive an allocated PCI device from kubelet. The following event is emitted:
+
+```
+Warning  FailedCreatePodSandBox  11s   kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "6be33fd689d91f280d6f82b9c6e3f018968fed39cd7762eb03f046bae10b8773": plugin type="multus" name="multus-cni-network" failed (add): [default/virt-launcher-testvmi-txhzt-p4hxv:sriov]: error adding container to network "sriov": SRIOV-CNI failed to load netconf: LoadConf(): VF pci addr is required
+```
+
+- In case of macvtap, macvtap cni fails for a similar reason: It cannot find a device because it does not receive an allocated device from kubelet. The following event is emitted:
+
+```
+Warning  FailedCreatePodSandBox  0s (x6 over 72s)  kubelet  : Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create pod network sandbox k8s_virt-launcher-testvmi-vnmz7-rdm9r_kubevirt-test-default1_d3ea53de-aa74-4128-b91a-a94d9c126e89_0(1f24220d8d124bd8552c287f16e06a35fd5e510f1ed9a08890927aa93094fc4e): error adding pod kubevirt-test-default1_virt-launcher-testvmi-vnmz7-rdm9r to CNI network "multus-cni-network": plugin type="multus-shim" name="multus-cni-network" failed (add): CmdAdd (shim): CNI request failed with status 400: 'ContainerID:"1f24220d8d124bd8552c287f16e06a35fd5e510f1ed9a08890927aa93094fc4e" Netns:"/var/run/netns/b5eaba98-9d38-42bc-b1d2-c2a101b65a90" IfName:"eth0" Args:"IgnoreUnknown=1;K8S_POD_NAMESPACE=kubevirt-test-default1;K8S_POD_NAME=virt-launcher-testvmi-vnmz7-rdm9r;K8S_POD_INFRA_CONTAINER_ID=1f24220d8d124bd8552c287f16e06a35fd5e510f1ed9a08890927aa93094fc4e;K8S_POD_UID=d3ea53de-aa74-4128-b91a-a94d9c126e89" Path:"" ERRORED: error configuring pod [kubevirt-test-default1/virt-launcher-testvmi-vnmz7-rdm9r] networking: [kubevirt-test-default1/virt-launcher-testvmi-vnmz7-rdm9r/d3ea53de-aa74-4128-b91a-a94d9c126e89:net1]: error adding container to network "net1": failed to lookup device "": Link not found
+```
+
+## Functional Testing Approach
+
+Existing e2e tests already rely on mapping of custom resources and thus will validate regression.
+
+## Implementation History
+
+<!--
+For example:
+01-02-1921: Implemented mechanism for doing great stuff. PR: <LINK>.
+03-04-1922: Added support for doing even greater stuff. PR: <LINK>.
+-->
+
+## Deprecation Requirements
+
+### Alpha
+
+  Will be introduced in KubeVirt v1.8, protected by the `DisableNADResourceInjection` FG. If the FG is set, KubeVirt will **not** map custom-resources.
+  Documentation will introduce the FG, highlight the deprecation and reference the network-resources-injector installation instructions.
+  A warning will be issued for the VM API in case secondary networks exist and FG is not enabled. A validation webhook will be implemented to issue the warning at VM creation. 
```

Comment:
I suggest the warning to be introduced in the Beta stage. Alpha is too early.
If this jumbs to Beta directly (with proper reasoning), it makes sense.